### PR TITLE
Lint shellscripts from tools

### PR DESCRIPTION
**Pull request by jml**
Related to https://github.com/weaveworks/build-tools/pull/22

Updates to use the version of tools that's in that PR, and adds a step to `lint` to run `shell-lint` over the `tools` directory. 

Doesn't change any of the shell scripts which are canonically located in scope, but does make sure that the tools changes don't break anything significant.

---
This change is [Reviewable](https://reviewable.io/reviews/weaveworks/scope/1658)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,6 +102,7 @@ tests: $(SCOPE_BACKEND_BUILD_UPTODATE)
 
 lint: $(SCOPE_BACKEND_BUILD_UPTODATE)
 	./tools/lint -ignorespelling "agre " -ignorespelling "AGRE " .
+	./tools/shell-lint tools
```
Comment (tomwilkie) [inline Makefile]: why not run it on the whole repo?

Comment (tomwilkie) [inline Makefile]: and why not make it part of lint?

Comment (jml) [inline Makefile]: > why not run it on the whole repo?

To limit the amount of work I'm doing at once. Goal of this PR isn't to verify the shell script in scope, but rather to prove that the shell changes in weaveworks/build-tools#22 actually work.

Fixing scope's shell script is a separate PR.

> and why not make it part of lint?

Once we've fixed shell scripts in all of our repos. Having it here lets us ratchet it up incrementally without having to do one big 'fix everything' push.

**Comment by tomwilkie** [comment]
LGTM

**Pull request merged**